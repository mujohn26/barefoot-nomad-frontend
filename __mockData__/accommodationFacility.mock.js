import {
	GET_ACCOMMODATION_SUCCES,
	GET_LOCATIONS_SUCCES,
	CREATE_ACCOMMODATION_SUCCES,
	GET_ACCOMMODATION_TYPES_SUCCES,
} from '../src/actions/accommodationFacilitiesActions';

export const props = {
	locations: [
		{ id: 1, city: 'kigali' },
		{ id: 2, city: 'kigali' },
		{ id: 3, city: 'kampala' },
	],
	RoomImagesResult: [{ Image: 'hello' }],
	amenities: [{ amenityName: 'swimming' }, { amenityName: 'parking' }],
	services: [{ serviceName: 'breakfast' }, { serviceName: 'cattering' }],
	result: [
		{
			name: 'marriot',
			description: 'five star hotel',
			location: {
				city: 'kigali',
			},
			category: 'hotel',
			owner: 'John',
			id: 2,
		},
	],
	RoomImagesResult: [{ Image: 'image1', cardId:0 }, { Image: 'image2', cardId:1 }],
	accommodationTypes: [
		{ id: 1, name: 'one bed room' },
		{ id: 2, name: 'two bed room' },
	],
	classes: {
		root: '',
	},
	rooms: [
		{
			numberOfRoom: '',
			typeId: '',
			price: '',
			currency: '',
			roomTypeImageUrl: '',
		},
		{
			numberOfRoom: '',
			typeId: '',
			price: '',
			currency: '',
			roomTypeImageUrl: '',
		}],
	images: [{ key:2, imageUrl: 'image1' }, { key:1,imageUrl: 'image2' }],
	index: 1,
	open: true,
	data: 'accommodation created successfully',
	handleDeleteCard: jest.fn(),
	handleChangeRooms: jest.fn(),
	handleChange: jest.fn(),
	getAccommodations: jest.fn(),
	getAccommodationsSuccess: jest.fn(),
	getLocations: jest.fn(),
	getlocationsSuccess: jest.fn(),
	uploadNewImageOnCloud: jest.fn(),
	getAccommodationType: jest.fn(),
	getaccommodationTypesSuccess: jest.fn(),
	uploadNewRoomImageOnCloud: jest.fn(),
	createAccomodationFacility: jest.fn(),
	createAccommodationSuccess: jest.fn(),
	cardClicked: jest.fn(),
	handleDeleteServiceChipCard: jest.fn(),
  handleDeleteAmenityChipCard: jest.fn(),
  handleDelete:jest.fn(),
  handleDeleteAccommodation:jest.fn(),
  handleDeleteRoomChipCard:jest.fn()
};

export const getAccommodations = {
	type: GET_ACCOMMODATION_SUCCES,
	payload: [
		{ id: 12, name: 'marriot' },
		{ id: 16, name: 'serena' },
	],
};

export const getLocations = {
	type: GET_LOCATIONS_SUCCES,
	payload: [
		{ id: 12, city: 'kigali' },
		{ id: 16, city: 'nairobi' },
	],
};
export const getAccommodationType = {
	type: GET_ACCOMMODATION_TYPES_SUCCES,
	payload: [
		{ id: 12, name: 'one bed room' },
		{ id: 16, name: 'two bed room' },
	],
};
export const uploadAccommodationImage = {
	type: 'UPLOAD_IMAGE_SUCCESS',
	attribute: [{ Image: 'image1' }],
};
export const uploadRoomImage = {
	type: 'UPLOAD_ROOM_IMAGE',
	attribute: [{ Image: 'image1' }],
};
export const createAccommodation = {
	type: CREATE_ACCOMMODATION_SUCCES,
	payload: 'accomodation created successfully',
};
export const isLoading = {
	type: 'LOADING',
	payload: true,

		};
export const getCardClicked = {
	type: 'GET_CARD_CLICKED',
	payload: 0,
};

