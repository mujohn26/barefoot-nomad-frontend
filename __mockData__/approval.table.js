import {
	GET_TRIP_REQUESTS_SUCCESS,
	GET_TRIP_REQUESTS_FAIL,
	SET_SELLECTED_TRIP,
	EDIT_TRIPREQUEST_STATUS_SUCCESS,
	EDIT_TRIPREQUEST_STATUS_FAIL,
	MAKE_OPEN_MODEL_TO_FALSE,
} from '../src/actions/approval.table';

export const props = {
	closeErrorMessageAlert: jest.fn(),
	popupState: true,
	GetNotifications: jest.fn(),

	getTripRequests: jest.fn(),
};

export const makeOpenModelToFalse = {
	type: MAKE_OPEN_MODEL_TO_FALSE,
	successMessage: false,
	errorMessage: false,
	message: '',
};
export const editTripRequestStatusFail = {
	type: EDIT_TRIPREQUEST_STATUS_FAIL,
	errorMessage: true,
	message: '',
};
export const editTripRequestStatusSuccess = {
	type: EDIT_TRIPREQUEST_STATUS_SUCCESS,
	successMessage: true,
	message: '',
};
export const setSelectedTripRequest = {
	type: SET_SELLECTED_TRIP,
	payload: {},
};
export const getTripRequestsFail = {
	type: GET_TRIP_REQUESTS_FAIL,
	tripRequestsError: false,
};
export const getTripRequestsSuccess = {
	type: GET_TRIP_REQUESTS_SUCCESS,
	tripRequests: { requestTrips: [[], []] },
};
export const token =
	'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImVtYWlsIjoibXVqb2huMjVAZ21haWwuY29tIiwiaXNWZXJpZmllZCI6dHJ1ZSwiaWQiOjMsInJvbGUiOiJyZXF1ZXN0ZXIifSwiaWF0IjoxNTgzOTI0ODAyLCJleHAiOjE1ODQwMTEyMDJ9.-9UymF5sdQOnZSTGL9Gnm1rIMnpHEGNxyeYl9X3_GmM';
