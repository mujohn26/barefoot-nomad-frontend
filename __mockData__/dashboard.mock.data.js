
export const props = {
	locations: 
			  [
				{
                    country: "Congo",
                    city: "Kinshasa",
                    travelledTimes: "7"
				},
             ],
             classes: {
                TypographyTitle: '',
            },
            isLoading:false,
getMostTraveledDestinations: jest.fn(),
useSelector:jest.fn()

};

export const getMostTraveledDestinations = {
	type: 'GET_MOST_TRAVELED_DESTINATION',
	payload: [
		{
            country: "Congo",
            city: "Kinshasa",
            travelledTimes: "7"
		},
	]
  };


 