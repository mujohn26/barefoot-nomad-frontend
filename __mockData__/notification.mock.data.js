import {
	MARK_NOTIFICATIONAS_READ,
	GET_NOTIFICATIONS_SUCCESS,
} from '../src/actions/notificationPane';

export const props = {
	classes: { textFild: '' },
	modal: true,
	closeErrorMessageAlert: jest.fn(),
	socketManagement: jest.fn(),
	// popupState: jest.fn(),
	popupState: true,
	GetNotifications: jest.fn(),
	UNSAFE_componentWillMount: jest.fn(),
	markNotificationAsRead: jest.fn(),
	notificationPaneHeigthHandle: jest.fn(),
	history: { push: jest.fn() },
	notifications: [{ read: true }, { read: true }],
	notification: [{ read: false }, { read: false }],
	notificationPaneHeigthHandle: jest.fn(),
	notificationGreaterThanNine: [
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
		{ read: false },
	],
};

export const socketNotification = {
	type: 'GET_SOCKET_NOTIFICATIONS',
	Notifications: {},
};
export const getNotificationsFail = {
	type: 'GET_NOTIFICATIONS_FAIL',
	NotificationError: {},
};
export const markNotificationAsReadFail = {
	type: 'MARK_NOTIFICATIONAS_READ_FAIL',
	markNotificationAsReadErrorMessage: {},
	open: '',
};
export const openErrorModel = {
	type: 'MAKE_OPEN_MODEL_TO_FALSE',
	open: false,
};

export const initialState = {
	Notifications: [],
	markNotificationAsRead: [],
	NotificationError: '',
	markNotificationAsReadErrorMessage: false,
	open: false,
};
export const getNotifications = {
	type: GET_NOTIFICATIONS_SUCCESS,
	Notifications: [
		{
			message: 'test message',
		},
		{
			message: 'test message',
		},
		{
			message: 'test message',
		},
	],
};

export const markNotificationAsRead = {
	type: MARK_NOTIFICATIONAS_READ,
	markNotificationAsReadMessage: {
		message: 'shema has updated a trip request',
	},
};

export const token =
	'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImVtYWlsIjoicmljc2hhbWFAZ21haWwuY29tIiwiaXNWZXJpZmllZCI6dHJ1ZSwiaWQiOjEsInJvbGUiOiJyZXF1ZXN0ZXIifSwiaWF0IjoxNTgyMzk1NjAyLCJleHAiOjE1ODI0ODIwMDJ9.idrtqvBO6UkVxqU32mW6XKC2a4XcltwETQtMWOP3lOQ';
