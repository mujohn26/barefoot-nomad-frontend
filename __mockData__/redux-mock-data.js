import {
	CHANGE_STATE,
	UPDATE_USER_PROFILE,
	GET_USER_PROFILE,
	CANCEL_USER_UPDATE,
	HTTP_REQUEST_START,
	UPLOAD_IMAGE_SUCCESS,
	ERROR_UPLOAD_IMAGE,
} from '../src/actions/user.profile.action';
export const props = {
	classes: { textFild: '' },
	UpdateduserProfileInfo: { successMessage: 'true' },
	socketManagement: jest.fn(),
	userProfileInfo: {},
	goodBye: jest.fn(),
	greeting: jest.fn(),
	changeAttribute: jest.fn(),
	GetUserProfile: jest.fn(),
	UNSAFE_componentWillMount: jest.fn(),
	changeAttribute: jest.fn(),
	updateUserProfile: jest.fn(),
	handleChange: jest.fn(),
	autoCompliteHandleChange: jest.fn(),
	switchHandleChange: jest.fn(),
	cancelUserUpdate: jest.fn(),
	uploadNewImageOnCloud: jest.fn(),
};

export const changeAttribute = {
	type: CHANGE_STATE,
	attribute: {
		firstName: '',
	},
};
export const startLoader = {
	type: HTTP_REQUEST_START,
	attribute: {
		profileImage: '',
	},
};

export const uploadUserImage = {
	type: UPLOAD_IMAGE_SUCCESS,
	attribute: {
		profileImage: '',
	},
};
export const FailuploadUserImage = {
	type: ERROR_UPLOAD_IMAGE,
	attribute: {
		uploadImageError: '',
	},
};

export const updateUserProfile = {
	type: UPDATE_USER_PROFILE,
	updatedProfile: {
		email: '',
		firstName: '',
		lastName: '',
		country: '',
		gender: '',
		birthdate: '',
		preferredlanguage: '',
		preferredcurrency: '',
		place: '',
		department: '',
		profileImage:
			'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png',
		appNotification: false,
		emailNotification: false,
		successMessage: '',
	},
};

export const GetUserProfile = {
	type: GET_USER_PROFILE,
	userProfileInfo: {
		email: '',
		firstName: '',
		lastName: '',
		country: '',
		gender: '',
		birthdate: '',
		preferredlanguage: '',
		preferredcurrency: '',
		place: '',
		department: '',
		profileImage:
			'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png',
		appNotification: false,
		emailNotification: false,
		successMessage: '',
	},
};

export const GetUserProfileReducer = {
	type: GET_USER_PROFILE,
	userProfileInfo: {
		data: {
			email: '',
			firstName: '',
			lastName: '',
			country: '',
			gender: '',
			birthdate: '',
			preferredlanguage: '',
			preferredcurrency: '',
			place: '',
			department: '',
			profileImage:
				'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png',
			appNotification: false,
			emailNotification: false,
			successMessage: '',
		},
	},
};
export const cancelUserUpdate = {
	type: CANCEL_USER_UPDATE,
	UpdateduserProfileInfo: {
		email: '',
		firstName: '',
		lastName: '',
		country: '',
		gender: '',
		birthdate: '',
		preferredlanguage: '',
		preferredcurrency: '',
		place: '',
		department: '',
		profileImage:
			'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png',
		appNotification: false,
		emailNotification: false,
		successMessage: '',
	},
};

export const initialState = {
	userProfileInfo: {
		email: '',
		firstName: '',
		lastName: '',
		country: '',
		gender: '',
		birthdate: '',
		preferredlanguage: '',
		preferredcurrency: '',
		place: '',
		department: '',
		profileImage:
			'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png',
		appNotification: false,
		emailNotification: false,
		successMessage: '',
	},
	UpdateduserProfileInfo: {
		email: '',
		firstName: '',
		lastName: '',
		country: '',
		gender: '',
		birthdate: '',
		preferredlanguage: '',
		preferredcurrency: '',
		place: '',
		department: '',
		profileImage:
			'https://res.cloudinary.com/dby88h516/image/upload/v1580575657/download_mimbni.png',
		appNotification: false,
		emailNotification: false,
		successMessage: '',
	},
	error: [],
};

export const data = {
	firstName: 'shema',
};
