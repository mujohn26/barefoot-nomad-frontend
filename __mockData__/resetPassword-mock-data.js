import {RESET_PASSWORD_SUCESS,FORGOT_PASSWORD_SUCCESS,FORGOT_PASSWORD_ERROR,RESET_PASSWORD_ERROR,DELETE_ERROR } from '../src/actions/resetPasswordAction';

export const props = {
	stateObject: {
		passwordReducer: {
			message: 'user not found!',
		},
	},
	classes: {
		paper:""
	},
	forgotMessage:'Email sent please ',
	resetMessage:'Password has successfuly changed',
	forgotMessageError:'user not found',
	state:{
		email:'mujohn25@gmail.com'
	},
	
	sendResetLink:jest.fn(),
	forgotPassword:jest.fn(),
	forgotPasswordError:jest.fn(),
	deleteForgotError:jest.fn(),
	resetPassword:jest.fn(),
	deleteResetError:jest.fn()
	
};
export const forgotEmailInputAction = {
	type: FORGOT_PASSWORD_SUCCESS,
	payload: 'reset link sent',
  };
  export const resetPasswordInputAction = {
	type:  RESET_PASSWORD_SUCESS,
	payload:{  
	status: 200,
    message: "Password has successfuly changed",
    data: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImVtYWlsIjoibXVqb2huMjZAZ21haWwuY29tIiwiaXNWZXJpZmllZCI6dHJ1ZSwiaWQiOjJ9LCJpYXQiOjE1ODA5MzczNTksImV4cCI6MTU4MTAyMzc1OX0.3IUkdzh7WxLyLYBi6hYNke_9ySu3l81yD7YR5mqCiYA",}
  };
  export const forgotPasswordInputActionError = {
	type: FORGOT_PASSWORD_ERROR,
	message:'user not found'
  };
  export const resetPasswordInputActionError = {
	type: RESET_PASSWORD_ERROR,
	message: [
		   "Password should be provided and must be alphanumeric with atleast 8 charactors.",
		    "conform Password should be provided and must be alphanumeric with atleast 8 charactors."
	            ]
  };
  export const deleteForgotErrorAction = {
	type: DELETE_ERROR,
	message: '',
  };
  export const deleteResetErrorAction = {
	type:'DELETE_RESET_ERROR',
	resetMessageError: ''
  };
  export const isLoadingAction = {
	type: 'LOADING',
	isLoading:true
  };
