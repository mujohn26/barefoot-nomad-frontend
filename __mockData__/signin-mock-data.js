export const props = {
    signInReducer: {
        user: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
        error: 'user not found!',
    },
    appReducer: {
        isLoading: false,
    },
    user: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
    message: {
        loggingIn: {
            signInReducer: {
                user: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
                error: 'user not found!',
            },
        },
    },
	stateObject: {
		signInReducer: {
            user: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
			error: 'user not found!',
		},
    },

	classes: {
        paper: "",
        form: "",
        submit: "",
        link: "",
        socialauth: "",
        aanchor_tags_class: ""
    },
    signIn: jest.fn(),
    logout: jest.fn()

};
export const success = {
	type: 'USER_SIGNIN_SUCCESS',
	user: {status: 200, message: "user has logged in successfully", data: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…2OTl9.qXCLeAzSBOBN3AYa0x6t-Ncisbo2qX3me6lZquJWHfM"},
  };
export const failure = {
	type: 'USER_SIGNIN_FAILURE',
	error: {status: 404, error: "Email or password does not match"},
  };
export const logout = {
    type: 'USER_LOGOUT'
}
