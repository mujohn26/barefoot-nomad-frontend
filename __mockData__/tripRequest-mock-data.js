export const props = {
	tripRequestReducer: {
		user:
			'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
		error: 'user not found!',
	},
	tripRequestsReducers: {
		tripToEdit: '',
	},
	appReducer: {
		isLoading: false,
	},
	user:
		'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
	info: {
		tripRequestReducer: {},
		locationsInfo: [
			{
				id: '10',
				city: 'kigali',
				country: 'rwanda',
			},
			{
				id: '12',
				city: 'bujumbura',
				country: 'burundi',
			},
		],
	},
	tripCreated: 'message',
	message: 'succes',
	stateObject: {
		signInReducer: {
			user:
				'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
			error: 'user not found!',
		},
	},

	classes: {
		paper: '',
		form: '',
		submit: '',
		link: '',
		socialauth: '',
		aanchor_tags_class: '',
	},
	value: {
		To: '',
		From: '',
		departureDate: '',
		reasons: '',
		socialauth: '',
		aanchor_tags_class: '',
		accomodationId: '',
	},
	accomodationInfomation: [{ name: 'marriot', id: '10', averageRate: 3 }, {}],
	accomodationInfo: {
		accommodations: [{ name: '', id: '10', averageRate: 3 }, {}],
	},
	screenSize: 'lg',
	onHandleChange: jest.fn(),
	handleChangeDateMultiCity: jest.fn(),
	handleDelete: jest.fn(),
	GetLocations: jest.fn(),
	GetAccomodations: jest.fn(),
	requestTrip: jest.fn(),
	onClick: jest.fn(),
	editTripRequest: jest.fn(),
};
export const props2 = {
	tripRequestReducer: {
		user:
			'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
		error: 'user not found!',
	},
	appReducer: {
		isLoading: false,
	},
	user:
		'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
	info: {
		tripRequestReducer: {},
		locationsInfo: [
			{
				id: '10',
				city: 'kigali',
				country: 'rwanda',
			},
			{
				id: '12',
				city: 'bujumbura',
				country: 'burundi',
			},
		],
	},
	tripCreated: 'message',
	message: 'succes',
	stateObject: {
		signInReducer: {
			user:
				'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkI…A0MH0.oMzzh3sFKmdHNp0izkG5aoiaKdn47GOg1vXho1ELy5Q',
			error: 'user not found!',
		},
	},

	classes: {
		paper: '',
		form: '',
		submit: '',
		link: '',
		socialauth: '',
		aanchor_tags_class: '',
	},
	value: {
		To: '10',
		From: '',
		departureDate: '',
		reasons: '',
		socialauth: '',
		aanchor_tags_class: '',
		accomodationId: '',
	},
	city: {},
	accomodationInfomation: [{ name: 'marriot', id: '10', averageRate: 3 }, {}],
	accomodationInfo: {
		accommodations: [{ name: '', id: '10', averageRate: 3 }, {}],
	},
	screenSize: 'lg',
	onHandleChange: jest.fn(),
	handleChangeDateMultiCity: jest.fn(),
	handleDelete: jest.fn(),
	GetLocations: jest.fn(),
	GetAccomodations: jest.fn(),
	requestTrip: jest.fn(),
	onClick: jest.fn(),
};
