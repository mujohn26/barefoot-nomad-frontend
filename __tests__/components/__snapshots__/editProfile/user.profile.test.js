import React from 'react';
import { shallow, mount } from 'enzyme';
import { props } from '../../../../__mockData__/redux-mock-data';
import {
	UserProfile,
	mapStateToProps,
} from '../../../../src/components/profile/userProfile';
import { JestEnvironment } from '@jest/environment';

describe('user profile component', () => {
	it('should render the user profile  component successfully', () => {
		const wrapper = shallow(<UserProfile {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should have initial state', () => {
		const initState = {
			userProfileReducer: { UpdateduserProfileInfo: '' },
			NotificationReducer: { Notifications: [] },
		};
		expect(mapStateToProps(initState)).toBeDefined();
	});
	it('should handle onclick logout button', () => {
		const wrapper = shallow(<UserProfile {...props} />);
		wrapper.setProps({
			logout: jest.fn(),
		});
		wrapper.setState({
			logout: false,
		});
		wrapper
			.find('WithStyles(ForwardRef(ListItem))')
			.props()
			.onClick();
		expect(wrapper.state('logout')).toEqual(true);
	});
});
