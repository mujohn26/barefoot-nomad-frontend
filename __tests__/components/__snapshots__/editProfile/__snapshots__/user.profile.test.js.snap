// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user profile component should render the user profile  component successfully 1`] = `
<Fragment>
  <Styled(MuiBox)
    style={
      Object {
        "height": "100%",
      }
    }
    width="100%"
  >
    <Link
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/profile"
    >
      <Styled(MuiBox)
        height="80px"
        width="100%"
      >
        <WithStyles(ForwardRef(Typography))
          align="center"
        >
          <img
            alt=""
            style={
              Object {
                "borderRadius": "50%",
                "height": "70px",
                "objectFit": "cover",
                "width": "30%",
              }
            }
          />
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <Styled(MuiBox)
        alignSelf={3}
        fontStyle={2}
      >
        <WithStyles(ForwardRef(Typography))
          align="center"
          style={
            Object {
              "color": "gray",
              "fontSize": "18px",
            }
          }
        >
          undefined undefined
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <Styled(MuiBox)
        marginTop={1}
      >
        <WithStyles(ForwardRef(Typography))
          align="center"
          style={
            Object {
              "color": "gray",
              "fontSize": "18px",
            }
          }
        />
      </Styled(MuiBox)>
    </Link>
  </Styled(MuiBox)>
  <Connect(WithStyles(withRouter(NavLinks))) />
  <Styled(MuiBox)
    marginTop={1}
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      id="dashboard"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <ExitToAppIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))>
        <WithStyles(ForwardRef(Typography))
          style={
            Object {
              "fontSize": "16px",
            }
          }
        >
          Logout
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(ListItemText))>
    </WithStyles(ForwardRef(ListItem))>
  </Styled(MuiBox)>
   
</Fragment>
`;
